src/loom/derived.clj

```diff
@@ -0,0 +1,60 @@
+(ns ^{:doc "Defines derived graphs from existing graphs using maps and filters."
+      :author "Horst Duchene"}
+  loom.derived
+  (:require [loom.graph :refer [digraph graph
+                                nodes edges successors fly-graph predecessors
+                                add-nodes* add-edges*
+                                directed?]]))
+
+(defn mapped-by
+  "Return a Graph or a DiGraph which has as nodeset (set (map f (nodes g)). An
+  edge [uu, vv] is an edge in the result iff g has an edge [u, v] such that [uu,
+  vv] = [(f u), (f v)]."
+  [f g]
+  (-> (if (directed? g) (digraph) (graph))
+      (add-nodes* (map f (nodes g)))
+      (add-edges* (map #(map f %) (edges g)))))
+
+(defn subgraph-starting-from
```

Comment:
Let's rename this to `subgraph-reachable-from` since `starting-from` behavior is ambiguous in directed graphs (could be bfs or dfs).